pkg/utils/logger/logger.go

```diff
@@ -13,69 +13,114 @@
+
+	Infof(format string, args ...interface{})
+
+	Info(format string)
+
+	Warnf(format string, args ...interface{})
+
+	Warn(format string)
+
+	Errorf(format string, args ...interface{})
+
+	Error(format string)
+
+	Fatalf(format string, args ...interface{})
+
+	Panicf(format string, args ...interface{})
+
+	WithFields(keyValues Fields) Logger
 }
 
-// SetupLogger sets up a file logger
-func SetupLogger(logFilePath string) {
-	logger, err := log.LoggerFromConfigAsString(fmt.Sprintf(logConfigFormat, getLogLevel(), getLogOutput(logFilePath)))
-	if err != nil {
-		fmt.Println("Error setting up logger: ", err)
-		return
-	}
-	err = log.ReplaceLogger(logger)
-	if err != nil {
-		fmt.Println("Error replacing logger: ", err)
-		return
-	}
+//Debugf formats its arguments according to the format, analogous to fmt.Printf,
+// and records the text as a log message at Debug level.
+func Debugf(format string, args ...interface{}) {
+	log.Debugf(format, args...)
 }
 
-func getLogLevel() string {
-	seelogLevel, ok := log.LogLevelFromString(strings.ToLower(os.Getenv(envLogLevel)))
-	if !ok {
-		seelogLevel = log.DebugLvl
-	}
-	return seelogLevel.String()
+//Debug formats its arguments according to the format, analogous to fmt.Debug,
+func Debug(format string) {
+	log.Debug(format)
 }
 
-func getLogOutput(logFilePath string) string {
-	switch logFilePath {
-	case "stdout":
-		return `<console />`
-	default:
-		return fmt.Sprintf(`<rollingfile filename="%s" type="date" datepattern="2006-01-02-15" archivetype="none" maxrolls="24" />`, logFilePath)
+// Infof is like Debugf, but at Info level.
+func Infof(format string, args ...interface{}) {
+	log.Infof(format, args...)
+}
+
+// Info is like Debugf, but at Info level.
+func Info(format string) {
+	log.Info(format)
+}
+
+// Warnf is like Debugf, but at Warning level.
+func Warnf(format string, args ...interface{}) {
+	log.Warnf(format, args...)
+}
+
+// Warn is like Debugf, but at Warning level.
+func Warn(format string) {
+	log.Warn(format)
+}
+
+// Errorf is like Debugf, but at Error level.
+func Errorf(format string, args ...interface{}) {
+	log.Errorf(format, args...)
+}
+
+// Error is like Debugf, but at Error level.
+func Error(format string) {
+	log.Error(format)
+}
+
+// Fatalf is like Debugf, but at Critical level and the logger then calls os.Exit(1).
+func Fatalf(format string, args ...interface{}) {
+	log.Fatalf(format, args...)
+}
+
+//Panicf is like Debugf, but at Panic level.
+func Panicf(format string, args ...interface{}) {
+	log.Panicf(format, args...)
+}
+
+//WithFields - If you want multiple fields
+func WithFields(keyValues Fields) Logger {
+	return log.WithFields(keyValues)
+}
+
+// GetZapLogger returns an instance of the logger, initializing a new logger
+func GetZapLogger() Logger {
+	if log == nil {
+		log := defaultZapLogger()
+		return &structuredLogger{
+			zapLogger: log,
+		}
 	}
+	return log
+}
+
+//InitZapLogger log  name generated by srvName
+func InitZapLogger(srvName string) Logger {
```

Comment:
A more idiomatic name for this function would be just `New()`.

Don't leak implementation details (the fact that we're using the zap library) out of the primary public interface. There's no reason for callers to know or care that we're using zap behind the scenes.